### PR TITLE
suppresses show-more facet links for bucketed range constraints

### LINKED ISSUE #74: Show more ignores buckets..
To be expected I guess, given the implementation, but I have a bucketed facet, and I show the show more. If I click it, I get extra values, but those contain individual values, not more buckets.

I guess the only way to make that work is call /v1/search for that particular facet with an increased limit, instead of calling /v1/values..

### PR DESCRIPTION
**Pull request by joemfb**
closes #74

### CODE REVIEW

**Comment by grtjn** [comment]
I think you can also prevent the show more link upfront, somewhere here: https://github.com/joemfb/ml-search-ng/blob/bucket-show-more/src/ml-search.service.js#L811. Bucketed facets have a distinct type:

```
		"updated": {
			"type": "bucketed",
			"facetValues": [{
				"name": "week ago",
				"count": 2,
				"value": "week ago"
			}, {
				"name": "month ago",
				"count": 4,
				"value": "month ago"
			}, {
				"name": "year ago",
				"count": 6,
				"value": "year ago"
			}, {
				"name": "longer",
				"count": 1,
				"value": "longer"
			}]
		},
```

**Comment by joemfb** [comment]
You're right; I had mistakenly thought that bucketed facets showed the type of the underlying index ...

I changed the existing check to an error / promise rejection, and added a check for each facet object. I still need another test for `ml-facets` ...

**Comment by grtjn** [comment]
Silly me, was pointing to the complete wrong line..

File: src/directives/ml-facets.directive.js [hunk 1]
```diff
@@ -61,6 +61,14 @@
     $scope.truncateLength = parseInt(attrs.truncate) || 20;
     $scope.shouldShowMore = !!attrs.showMore;
     $scope.shouldNegate = !!attrs.negate && !!attrs.activeFacets;
+
+    $scope.$watch('facets', function(newVal, oldVal) {
+      _.each(newVal, function(facet) {
+        if ( facet.type === 'bucketed' ) {
+          facet.displayingAll = true;
```
Comment (grtjn) [inline src/directives/ml-facets.directive.js]: Sorry, was pointing to wrong line. I meant to say, you could also toggle this flag when you receive the data in the service, where metadata and such is rewritten..

Comment (joemfb) [inline src/directives/ml-facets.directive.js]: That's a good point, I'll push an update.

**Comment by joemfb** [comment]
It's a good call, much simpler this way. I'll add a test confirming it's working as expected.

**Comment by grtjn** [comment]
I'll see if I can run a test with this as well, my github-search has bucketed facets, just need to drop in you branch..

**Comment by grtjn** [comment]
Whoops, not as easy to test as I thought. The bower.json points to dist/, and can't build ml-search-ng myself..

**Comment by joemfb** [comment]
You'll have to build it locally, than copy `dist/` into your bower_components directory.

Something like:

    git fetch upstream bucket-show-more
    git checkout -b bucket-show-more upstream/bucket-show-more

    gulp
    cp dist/* $project/bower_components/ml-search-ng/dist

**Comment by grtjn** [comment]
Manually copied the changes, that seemed to work nicely..

**Comment by grtjn** [comment]
Oh, I thought I didn't have the build stuff, but I do. Duh.. I have tested manually, I think this looks good..

**Comment by joemfb** [comment]
I've added a test for the bucketed facet annotation. Between that and your test, I'm comfortable merging.

**Pull request merged**